app/src/main/java/org/mozilla/fenix/components/Components.kt

```diff
@@ -222,7 +234,8 @@ class Components(private val context: Context) {
                 PocketUpdatesMiddleware(
                     core.pocketStoriesService,
                     context.pocketStoriesSelectedCategoriesDataStore
-                )
+                ),
+                MessagingMiddleware(messagingStorage = messagingStorage)
```

Comment:
There is one thing we haven't considered yet, which is startup performance: we are now initializing and using experiments from the beginning. Even if we made this storage lazy, we _want_ messaging to work at cold boot I assume (right?), so the future of the `KeyPairMessageMetadataStorage` and how the messaging feature inits (off the main thread) might have an effect.

Reply:
> Even if we made this storage lazy, we want messaging to work at cold boot I assume (right?)

Yeah, ideally. But a guess we have the same restrictions as normal experiments have, I think it's OK that messages will be shown after the second startup run.